vault/identity_store_util.go

```diff
@@ -223,7 +277,9 @@ func (i *IdentityStore) upsertEntityInTxn(ctx context.Context, txn *memdb.Txn, e
 		return fmt.Errorf("entity is nil")
 	}
 
-	for _, alias := range entity.Aliases {
+	aliasFactors := make([]string, len(entity.Aliases))
```

Comment:
It seems that `aliasFactors` gets populated but not used by anything outside of the for loop.

Reply:
True, it is only used inside the `for` loop.

Reply:
Is it more efficient to pre-allocate the slice or does the index/location of the entry matter in the particular case? Wondering why we do index-based assignment vs `append` in https://github.com/hashicorp/vault/pull/5404/files/750917dbe1936ea2ff7baf74d28cf4214cc0396a#diff-24216d724142bf1f2526e49f330d905bR314.

Reply:
It seems that assignment is indeed more efficient.

Reply:
Yeah. Append does resizing of the underlying array multiple times.